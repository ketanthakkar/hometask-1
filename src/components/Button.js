import React, { Component } from 'react';
import '../styleModules/Button.css';

const Button = ({name, active, handleClick, id}) => {
    return (
        <button 
            className = {`button ${active === id ? 'button--active' : 'button--unactive'}`} 
            onClick = {handleClick}
            id = {id}
        >
            {name} 
        </button>
    );
}
export default Button;
