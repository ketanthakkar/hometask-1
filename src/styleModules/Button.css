:root {
    --button-active:pink;
    --button-unactive: black;
}
.button__active{
    background:var(--button-active);
    border: solid var(--button-active);
    width:70px;
    height: 30px;
    color:white;
}

.button__unactive {
    background:var(--button-unactive);
    border: solid var(--button-unactive);
    width:70px;
    height: 30px;
    color:white;
}