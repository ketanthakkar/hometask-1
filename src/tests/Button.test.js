import React from 'react';
import { shallow } from 'enzyme';
import Button from '../components/Button';


describe('Button', () => {
    it('should render unactive button', () => {
      const render = shallow(
          <Button name = 'test' />
      );
    
      expect(render.find('button').hasClass('button--unactive')).toBe(true);
      expect(render).toMatchSnapshot();
    });

    it('should render active button', () => {
      const render = shallow(
        <Button name = 'test' active/>
      );
      expect(render.find('button').hasClass('button--active')).toBe(true);
      expect(render).toMatchSnapshot();
    })
    it('should not modify props', () => {
      const name = 'name';
      const active = false;
      const render = shallow(
        <Button name = {name} active = {active} />
      );
      expect(name).toEqual('name')
      expect(active).toEqual(false);
      expect(render).toMatchSnapshot();
    })
});