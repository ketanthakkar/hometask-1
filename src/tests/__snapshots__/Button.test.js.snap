// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should not modify props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    active={false}
    name="name"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "name",
      "className": "button button--unactive",
      "onClick": undefined,
    },
    "ref": null,
    "rendered": "name",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "name",
        "className": "button button--unactive",
        "onClick": undefined,
      },
      "ref": null,
      "rendered": "name",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Button should render active button 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    active={true}
    name="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "test",
      "className": "button button--active",
      "onClick": undefined,
    },
    "ref": null,
    "rendered": "test",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "test",
        "className": "button button--active",
        "onClick": undefined,
      },
      "ref": null,
      "rendered": "test",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Button should render unactive button 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    name="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "test",
      "className": "button button--unactive",
      "onClick": undefined,
    },
    "ref": null,
    "rendered": "test",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "test",
        "className": "button button--unactive",
        "onClick": undefined,
      },
      "ref": null,
      "rendered": "test",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
